### PR TITLE
AzureStack AzureConsistentStorage update for branch Release 1.2.4

### PR DESCRIPTION
**Pull request by eridai**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@eridai__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by markcowl** [comment]
@eridai Please follow the directions in http://aka.ms/azuregithub to assure that the cla bot doesn't flag you on every PR

File: src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage/Farm/SetFarm.cs [hunk 1]
```diff
@@ -67,11 +74,266 @@ public string FarmName
         [SettingField]
         public string CorsAllowedOriginsList { get; set; }
 
-        ///
+        /// <summary>
```
Comment (markcowl) [inline src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage/Farm/SetFarm.cs]: Let's have some actual documentation in the summary comments for these

File: src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage/Fault/GetFaultHistory.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            set;
+        }
+
+        /// <summary>
+        ///     Resource Uri
+        /// </summary>
+        [Alias("Id")]
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNull]
+        public string ResourceUri
```
Comment (markcowl) [inline src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage/Fault/GetFaultHistory.cs]: Isn't this in a separate parameter set from Resource Group / Farm name?

File: src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage/StorageAccount/SyncStorageAccount.cs [hunk 3]
```diff
@@ -61,9 +68,15 @@ public sealed class SyncStorageAccount : AdminCmdlet
         /// </summary>
         [Parameter(Mandatory = false, Position = 8)]
         public string StorageAccountApiVersion { get; set; }
-        
+
+        /// <summary>
+        /// Specifies the Microsoft.Resource.Admin apiVersion
+        /// </summary>
+        [Parameter(Mandatory = false, Position = 9)]
```
Comment (markcowl) [inline src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage/StorageAccount/SyncStorageAccount.cs]: You probably don't want 9 positional parameters

File: src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage/StorageAccount/UndeleteStorageAccount.cs [hunk 4]
```diff
@@ -54,6 +61,12 @@ public sealed class UndeleteStorageAccount : AdminCmdlet
         [Parameter(Mandatory = false, Position = 7)]
         public string StorageAccountApiVersion { get; set; }
 
+        /// <summary>
+        /// Specifies the Microsoft.Resource.Admin apiVersion
+        /// </summary>
+        [Parameter(Mandatory = false, Position = 8)]
```
Comment (markcowl) [inline src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage/StorageAccount/UndeleteStorageAccount.cs]: too many positional parameters

**Comment by markcowl** [comment]
@eridai  A few questions about cmdlet interface, please respond

File: src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage/StorageAccount/SyncStorageAccount.cs (hunk 3 above)
Comment (eridai) [inline src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage/StorageAccount/SyncStorageAccount.cs]: So the suggestion is to remove the position attribute for this cmdlet?

File: src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage/Fault/GetFaultHistory.cs (hunk 2 above)
Comment (eridai) [inline src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage/Fault/GetFaultHistory.cs]: No. StartTime, EndTime and ResourceUri are in the same paramter set with Resource Group and Farm name. What is the concern here?

Comment (markcowl) [inline src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage/Fault/GetFaultHistory.cs]: So, does ResourceUri describe a different resource than the resource described by resource grup name + farm name?  I want to make sure we are not allowing / requiring the user to identify the same resource twice.

File: src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage/StorageAccount/SyncStorageAccount.cs (hunk 3 above)
Comment (markcowl) [inline src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage/StorageAccount/SyncStorageAccount.cs]: No more than 3 or 4 parameters should be positional.  This should include the required parameters and the most common optional parameters.  Other optional parameters should neo be positional

File: src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage/Fault/GetFaultHistory.cs (hunk 2 above)
Comment (eridai) [inline src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage/Fault/GetFaultHistory.cs]: Oh, I see. And yes, ResourceUri describe a different resource. It refers to the resource where fault happens. Ex. a specific VM node.

File: src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage/StorageAccount/SyncStorageAccount.cs (hunk 3 above)
Comment (eridai) [inline src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage/StorageAccount/SyncStorageAccount.cs]: I will remove the position attribute for new parameters while leaving the ones already exist. Sounds reasonable?

**Comment by markcowl** [comment]
@eridai Please respond to review comments.  This cannot be merged until this occurs.

**Comment by markcowl** [comment]
@eridai Merge conflicts please update

**Pull request closed without merging**